app/grandchallenge/evaluation/templates/evaluation/leaderboard_detail.html

```diff
@@ -14,14 +14,33 @@
 {% endblock %}
 
 {% block sidebar %}
-    <div class="nav nav-pills col-12 pl-3 mb-3">
+    <div class="nav nav-pills d-none {% if challenge.phase_set.count > 4 %}d-lg-inline-flex{% elif challenge.phase_set.count == 4 %}d-md-inline-flex{% else %}d-sm-inline-flex{% endif %} col-12 pl-3 mb-3">
         {% for phase in challenge.phase_set.all %}
             <li class="nav-item">
-                <a class="nav-link {% if request.resolver_match.view_name == 'evaluation:leaderboard' and request.resolver_match.kwargs.slug == phase.slug %}active{% endif %}"
+                <a class="nav-link mr-1 px-4 py-1 {% if request.resolver_match.view_name == 'evaluation:leaderboard' and request.resolver_match.kwargs.slug == phase.slug %}active{% endif %}"
                    href="{% url 'evaluation:leaderboard' challenge_short_name=challenge.short_name slug=phase.slug %}">{{ phase.title }}</a>
             </li>
         {% endfor %}
     </div>
+    <div class="nav nav-pills d-block {% if challenge.phase_set.count > 4 %}d-lg-none{% elif challenge.phase_set.count == 4 %}d-md-none{% else %}d-sm-none{% endif %} col-12 pl-3 mb-3">
```

Comment:
Have you checked for duplicate queries using the debug toolbar? You might be going to the database quite a few times here with a lot of instances of `phase_set.all` and `phase_set.count`. Check, and if so you can probably reduce the number of calls by using the challenge context processor.

Reply:
As far as I can tell, there are no duplicate database hits. Same number of queries with and without the if statement in there. I also thought that the count method does not actually evaluate the queryset and hence would not hit the database at all? Looking through the list of queries with the debug toolbar, I also can't find this query anywhere. I hope I'm not just missing it.